presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -1748,6 +1748,55 @@ public void testRollupOverUnion()
     }
 
     @Test
+    public void testGrouping()
+            throws Exception
+    {
+        assertQuery("" +
+            "SELECT a, b, sum(c), grouping(a, b) + grouping(a)\n" +
+            "FROM (VALUES\n" +
+            "      ('h', 'j', 11),\n" +
+            "      ('k', 'l', 7)\n" +
+            "      ) AS t (a, b, c)\n" +
+            "GROUP BY GROUPING SETS ( (a), (b))\n" +
+            "ORDER BY grouping(b) ASC",
+            "VALUES (NULL, 'j', 11, 3),\n" +
+                    "(NULL, 'l', 7, 3),\n" +
+                    "('h', NULL, 11, 1),\n" +
+                    "('k', NULL, 7, 1)");
+    }
+
+    @Test
+    public void testGroupingInWindowFunction()
```

Comment:
This test is inspired by Q36 of TPC-DS. I wonder if I should create product tests for this functionality given this test. What do you think? I think this test along with the other unit tests cover everything.